workers/unity/Packages/com.improbable.gdk.core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -67,6 +60,20 @@ private void GenerateComponentGroups()
                     typesToRemove.Add(type);
                     addRemoveComponentActionMethod.MakeGenericMethod(type).Invoke(this, null);
                 }
+
+                // Find all ComponentCleanupHandlers
+                foreach (var type in assembly.GetTypes())
+                {
+                    if (typeof(ComponentCleanupHandler).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var componentCleanupHandler = Activator.CreateInstance(type) as ComponentCleanupHandler;
+                        foreach (var componentType in componentCleanupHandler.CleanUpComponentTypes)
+                        {
+                            typesToRemove.Add(componentType.GetManagedType());
+                            addRemoveComponentActionMethod.MakeGenericMethod(componentType.GetManagedType()).Invoke(this, null);
```

Comment:
small sidenote that MakeGeneric doesnt work with IL2CPP, so would be nice to avoid where possible.

Reply:
Hmmm.. cc @samiwh might need to change this original code :/

Reply:
I think we consider this later. If we get rid of it completely there are a lot of features we can't make nearly as easily.